### PR TITLE
feat: add guest nickname handling

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/17147
Fix https://github.com/nextcloud-libraries/nextcloud-browser-storage/issues/3

### CODE REVIEW

File: lib/publicShare.ts [hunk 1]
```diff
@@ -5,6 +5,11 @@
  * @module public
  */
 import { loadState } from '@nextcloud/initial-state'
+import { getBuilder } from '@nextcloud/browser-storage'
+
+const localStorage = getBuilder('shared')
```
Comment (skjnldsv) [inline lib/publicShare.ts:10]: Or `getBuilder('public')` ?

File: lib/publicShare.ts [hunk 2]
```diff
@@ -5,6 +5,9 @@
  * @module public
  */
 import { loadState } from '@nextcloud/initial-state'
+import { getBuilder } from '@nextcloud/browser-storage'
+
+const localStorage = getBuilder('shared').persist().build()
```
Comment (ShGKme) [inline lib/publicShare.ts:10]: To not shadow global `localStorage` variable and keep a commonly used name.

```suggestion
const browserStorage = getBuilder('shared').persist().build()
```

**Review by ShGKme — APPROVED** [review]
Makes sense to fix both issues. But seems unrelated to sharing library, or do I get meaning of this library wrong?

File: lib/publicShare.ts (hunk 2 above)
Comment (skjnldsv) [reply lib/publicShare.ts:10]: didn't even think of that :sweat_smile:

**Comment by skjnldsv** [comment]
> But seems unrelated to sharing library, or do I get meaning of this library wrong?

Tough discussion. I think it fits tbh. We already have some public sharing handling here.
Every app that uses the Guest nickname would be in a context of sharing (talk conversation share link, file/folder share link, poll.... etc

So I think this is where it makes more sense :)
Not to be confused with the guest app here :warning:

**Comment by susnux** [comment]
Hm I am not that sure, as this library is more about files sharing (potentially we should add the Share model etc to this library).

For me this part should probably be in the same library as `getCurrentUser` so auth?

**Comment by susnux** [comment]
(code looks fine tough)

**Comment by skjnldsv** [comment]
> potentially we should add the Share model etc to this library

Our entire sharing backend  allows any resources to be shared, not only file :)
Theoretically. So any sharing related thing wuold belong here I would say :thinking: 

Nonetheless, I would also agree, I like the idea to have this on auth.
Since we're a few that will use that one, I'll let you all decide :eyes:

**Comment by juliushaertl** [comment]
Auth sounds even more fitting to me

**Comment by skjnldsv** [comment]
Let's gooo! The people have spoken! :judge:

**Comment by skjnldsv** [comment]
https://github.com/nextcloud-libraries/nextcloud-auth/pull/666

**Pull request closed without merging**